lib/objectLog.ts

```diff
@@ -0,0 +1,194 @@
+import { ToolingInterface } from '../drivers/instrumentedResource';
+import { logger } from './logger';
+import { promises as fs } from 'fs';
+import { getAwsConfig } from './awsConfig';
+import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { ActionAuditEntry } from '../actions/audit';
+
+abstract class ObjectLog {
+  protected readonly logger;
+  protected constructor() {
+    this.logger = logger;
+  }
+
+  abstract process(): void;
+}
+
+export interface DataTable {
+  header(): string[];
+  data(): string[][];
```

Comment:
I know it's a bit less space-efficient, but maybe this could be a list-of-object, and use the object-keys for the headers?